Feature/main page

doxliydriw opened the pull request:
Main page layout according to Figma project with functional components.

doxliydriw committed:
first commit. basic button setup. header and footer.

doxliydriw committed:
first commit

doxliydriw committed:
basicbutton, navbar and mainroute update

doxliydriw committed:
update to previous commit

doxliydriw committed:
section four finished

doxliydriw committed:
all elements according to figma

DimaShubin commented on src/components/Buttons/BasicButton/BasicButtonStyled.js:
```diff
@@ -0,0 +1,34 @@
+
+
+export const BasicButtonLayout = styled(Button)`
+&&& {
```
What does this "&&&" mean?

doxliydriw replied:
mistyped, should be:
export const BasicButtonLayout = styled(Button)`
&& {
in order to inherit Button styles from MUI components and add custom styles to it.

DimaShubin commented on src/components/Buttons/BasicButton/BasicButton.jsx:
```diff
@@ -0,0 +1,13 @@
+import { BasicButtonLayout } from '@/components/Buttons/BasicButton/BasicButtonStyled';
+
+const BasicButton = props => {
```
```suggestion
const BasicButton = ({ text, handleClick, variant, sx }) => {
```

doxliydriw replied:
thank you, did know it's possible to decompose in such way.

DimaShubin commented on src/components/Buttons/Bubble/Bubble.jsx:
```diff
@@ -0,0 +1,12 @@
+const Bubble = props => {
+  const { text, ...position } = props;
+  return (
+    <BubbleStyled left={position.left} marginbottom={position.marginbottom}>
```
Styled components custom props should start with '$' https://arc.net/l/quote/qzsatkhv

doxliydriw replied:
thank you, didn't knew that. fixed.

DimaShubin commented on src/components/NavigationBar/Footer.jsx:
```diff
@@ -0,0 +1,5 @@
+import NavBar from '@/components/NavigationBar/NavBar';
+
+const Header = () => <NavBar type="footer" />;
```
I guess it should be renamed.

doxliydriw replied:
fixed.

DimaShubin commented on src/components/NavigationBar/Footer.jsx:
```diff
@@ -0,0 +1,5 @@
+import NavBar from '@/components/NavigationBar/NavBar';
```
It is a bit confuising that footer and header are in the NavigatinBar folder.

doxliydriw replied:
fixed

DimaShubin commented on src/components/NavigationBar/MainPageNavBarStyled.js:
```diff
@@ -0,0 +1,54 @@
+      left: 0;
+        `
+      : `
+          position: abolute;
```
Misstyping. + You don't need to make it 'absolute' at all("Check footer styles in dev tools").

doxliydriw replied:
fixed

DimaShubin commented on src/components/NavigationBar/MainPageNavBarStyled.js:
```diff
@@ -0,0 +1,54 @@
+
+export const NavBarLayout = styled.nav`
+${(props) =>
+    props.type === 'header'
```
Magic string.. https://softwareengineering.stackexchange.com/questions/365339/what-is-wrong-with-magic-strings

doxliydriw replied:
renamed to navBarType.

DimaShubin commented on src/components/NavigationBar/NavBar.jsx:
```diff
@@ -0,0 +1,49 @@
+  const navigate = useNavigate();
+
+  const handleRegisterOpen = () => {
+    navigate('/register');
```
Use constants.

doxliydriw replied:
fixed.

DimaShubin commented on src/images/icons/Vector Globe Icon.svg:
```diff
@@ -0,0 +1,3 @@
+<svg width="53" height="52" viewBox="0 0 53 52" fill="none" xmlns="http://www.w3.org/2000/svg">
+<path fill-rule="evenodd" clip-rule="evenodd" d="M28.3102 0.0650931C28.0518 -0.00243305 27.7803 -0.0184745 27.5125 0.0208523C27.166 0.00697352 26.8187 -5.62993e-06 26.4707 -5.62993e-06C26.1228 -5.62993e-06 25.7756 0.00696923 25.4293 0.0208391C25.1615 -0.0184629 24.8901 -0.00243193 24.6318 0.0650521C18.4082 0.506082 12.5266 3.17454 8.08593 7.61522C3.20998 12.4912 0.470703 19.1044 0.470703 26C0.470 ...
```
It's better avoid using hardcoded colors for svg images. But let it be.

doxliydriw replied:
fixed. made svg as react component and passed styles with colours using styled-component.

DimaShubin commented on src/routes/MainRoute/MainRouteStyled.js:
```diff
@@ -0,0 +1,141 @@
+import { Typography, Box } from '@mui/material';
+
+export const MainPageLayout = styled(Box)`
+box-sizing: border-box;
```
Wrong indents. Seems like prettier plugin doesn't work for you.

doxliydriw replied:
Indents fixed. Something really wrong with prettier.

DimaShubin commented on src/routes/MainRoute/MainRoute.jsx:
```diff
@@ -7,20 +32,157 @@ const MainRoute = () => {
+      <MainPageSectionFour id="benefits">
+        <SectionFourContainer>
+          <Bubble
+            marginbottom="20px"
```
In otder to avoid such long repeatable content which is static also, you could create a varialble to store all the info you need as an array and render it with array method 'map' or something.

doxliydriw replied:
fixed.

DimaShubin commented on src/routes/MainRoute/MainRouteStyled.js:
```diff
@@ -0,0 +1,141 @@
+text-align: center;
+`
+/*
+export const MainPageTitle = styled(Typography)`
```
You can try to hook it off by using MUI classes if you like it more. https://arc.net/l/quote/xvfhbyax

doxliydriw replied:
You mean  not to create MainPageTitle but simply use 
<Typography sx={{
 fontSize: 20px;
  lineHeight: 24px;
  textAlign: center;
  marginBottom: 32px;
  }}
  in component?

DimaShubin commented on src/routes/MainRoute/MainRouteStyled.js:
```diff
@@ -0,0 +1,141 @@
+padding-top: 80px;
+font-family: Roboto, sans-serif;
+`
+
```
It is beeter to use ";" in the end of each styled component. It just make it easier to read and support the file itself.

doxliydriw replied:
Noted. fixed and will look why prettier stopped working on these files.

doxliydriw committed:
upd

doxliydriw committed:
upd

doxliydriw committed:
svg as components test

doxliydriw committed:
fixed accoring to comments except
https://github.com/ReginaVarybrus/talk-and-travel-chat/pull/21#discussion_r1572574762

DimaShubin commented on src/components/Buttons/BasicButton/BasicButton.jsx:
```diff
@@ -0,0 +1,11 @@
+    </BasicButtonLayout>
+  );
+};
+
```
```suggestion
const BasicButton = ({ text, handleClick, variant, sx }) => (
  <BasicButtonLayout sx={sx} onClick={handleClick} variant={variant}>
    {text}
  </BasicButtonLayout>
);

```

doxliydriw replied:
fixed

doxliydriw committed:
package-lock.json removed from pull request

DimaShubin commented on src/components/Buttons/BasicButton/BasicButton.jsx:
```diff
@@ -0,0 +1,11 @@
+import { BasicButtonLayout } from '@/components/Buttons/BasicButton/BasicButtonStyled';
+
+const BasicButton = ({ text, handleClick, variant, sx }) => {
```
Unexpected block statement surrounding arrow body; move the returned value immediately after the `=>`.

DimaShubin commented on src/routes/MainRoute/MainRoute.jsx:
```diff
@@ -1,26 +1,207 @@
-  const handleLoginOpen = () => {
-    navigate('/login');
-  };
+  const bubbleData = [
```
In order to render this static data every render you should better move it otside of the component.

doxliydriw replied:
fixed

DimaShubin commented on src/routes/MainRoute/MainRoute.jsx:
```diff
@@ -1,26 +1,207 @@
   const navigate = useNavigate();
-
   const handleRegisterOpen = () => {
     navigate('/register');
```
Use constant instead of magic string. (routesConfig.REGISTER)

doxliydriw replied:
fixed

DimaShubin commented on src/components/Header/Header.jsx:
```diff
@@ -0,0 +1,22 @@
+  const [isvisible, setIsVisible] = useState(true);
+  const [prevScrollPos, setPrevScrollPos] = useState(0);
+
+  useEffect(() => {
```
I would go with custom hook for such things..

doxliydriw replied:
fixed.

DimaShubin commented on src/components/Buttons/BasicButton/BasicButtonStyled.js:
```diff
@@ -0,0 +1,34 @@
+    padding: 12px 24px;
+
+    ${(props) =>
+    props.variant === 'outlined'
```
In order to avoid such magic string it would be better to move it to the compoennt. But let it be as is.

DimaShubin commented on src/routes/MainRoute/MainRouteStyled.js:
```diff
@@ -0,0 +1,133 @@
+  display: flex;
+  gap: 24px;
+  padding: 18.5px 120px 18.5px 0;
+  align-items: center
```
; missed

doxliydriw replied:
fixed.

doxliydriw committed:
fixed accoring to comments
